Assets/Scripts/Player/ShapeMatcher.cs

```diff
@@ -0,0 +1,47 @@
+using System.Collections;
+using System.Collections.Generic;
+using System.Linq;
+using UnityEngine;
+
+public class ShapeMatcher : MonoBehaviour
+{
+    public PilarCheck[] puzzlePieces;
+    public bool puzzlePiecesComplete;
+    public bool allPilarsInPlace;
+    bool allPuzzlePiecesInPlace = false;
+    public PilarCheck pilarCheck;
+
+    void Start()
+    {
+        pilarCheck = FindObjectOfType<PilarCheck>();
+        pilarCheck.pilarInPlace = false;
+    }
+
+    void Update()
+    {
+        if (puzzlePieces.All(x => x.pilarInPlace))
+        {
+            allPuzzlePiecesInPlace = true;
+        }
+        if (allPuzzlePiecesInPlace == pilarCheck.pilarInPlace == true)
+        {
+            allPilarsInPlace = true;
+        }
+        if (pilarCheck.pilarInPlace == true && allPilarsInPlace == true)
+        {
+            puzzlePiecesComplete = true;
+        }
+        if (puzzlePiecesComplete == true)
+        {
+            PuzzleComplete();
+        }
+        else
+        {
+            Debug.Log("puzzle not filled");
+        }
+    }
+    public void PuzzleComplete()
+    {
+        Debug.Log("Puzzle complete! Congrats!");
+    }
```

Comment:
```suggestion
    public void CheckCompletion()
    {
        if (pillars.All(x => x.Completed))
        {
            onCompletion?.Invoke()
        }
    }
```
Beter om niet de completion te checken in de update. deze method moet aangeroepen worden via een event elke keer als een pillar gecomplete word. hij checkt dan of de pillars completed zijn en dan roept hij event aan.